README.md

```diff
@@ -477,6 +504,7 @@ import net.openid.appauth.browser.VersionedBrowserMatcher
 */
 fun secureRedirection(uri: Uri) {
    val (basePackageName, baseCertFingerprints) = getAssetLinksJsonFile(uri)
+   // JJH: I think in the general case it's possible to have more than one package name match a given uri? https://stackoverflow.com/questions/48056006/how-to-create-assetlinks-json-when-website-is-handled-by-multiple-apps - so this should perhaps return an array? (I also can't find the source for this function)
```

Comment:
It's true that we can have more than one app per domain, but how would we choose which app to use? My goal is to provide one method that only takes a URI as a parameter to redirect the user from one app to another. I think this is not possible if we allow multiple apps per domain. Any ideas on how to handle this?

I did not add the source for the ``getAssetLinksJsonFile(uri)`` for brevity. I thought it's clear what the method does and it would only add a lot of code to the example.

Reply:
I think picking the first one that's actually installed and matches might be fine?

I guess the important case is the one mentioned on https://developer.android.com/training/app-links/verify-site-associations#multiple-apps - whilst there may be multiple apps, and the user might have all of them installed, only one of them might handle the path for the Authorization Endpoint, via a 'pathPrefix' filter as per https://developer.android.com/guide/topics/manifest/data-element

Reply:
For now I've added a comment in the code that the general case is more complex.

Reply:
Sounds good! I will integrate the check for multiple apps per domain into my API. The code will check which apps match the path and if there are multiple apps then it takes the first one that is installed.

Reply:
There is a problem with this approach to find the right app. The ``assetlinks.json`` file lets you only defined that an app is associated with the website but not which paths it handles. The paths are defined in the intent-filter of the app. So this means we cannot sort the apps for the paths they are handling and have to choose the first one that is installed. If this app cannot handle the URL it would be the job of this app to redirect the user to the correct app of this domain. I guess that is not optimal but we cannot change something about it. The same topic is discussed in this Stackoverflow discussion: 

https://stackoverflow.com/questions/35840262/handling-specific-routes-in-android-m-app-links

Reply:
Just to check what you're saying here - the problem is that we can't retrieve the intent-filter via the package manager API? (Implying that whatever implements this same logic when handling an app link is using a non-public API to retrieve that information?)

Reply:
You are right. I did not think this through. It is possible to check with the package manager which apps support an intent-filter. So we can intersect the set of installed apps with the set of apps that are listed in the ``assetlinks.json`` file and get a list of possible apps. I guess if there is more than one app left, we just choose the first one?

I just tested the retrieval of installed apps that support an intent-filter with the following code: 

```kotlin
private fun getAppsForIntent(): Set<String> {
    /*
        Source: https://stackoverflow.com/questions/11904158/can-i-disable-an-option-when-i-call-intent-action-view
     */
    val intent = Intent(Intent.ACTION_VIEW, Uri.parse("http://openidprovider.app2app.unsicher.ovh/c2id-login"))
    intent.addCategory(Intent.CATEGORY_DEFAULT)
    intent.addCategory(Intent.CATEGORY_BROWSABLE)

    val infos: List<ResolveInfo> = packageManager.queryIntentActivities(intent, PackageManager.GET_RESOLVED_FILTER)

    val packageNames = mutableSetOf<String>()
    for (info in infos) {
        val activityInfo = info.activityInfo
        println("Queried Package Name ${activityInfo.packageName}")
        packageNames.add(activityInfo.packageName)
    }
    return packageNames
}
```

Reply:
> I guess if there is more than one app left, we just choose the first one?

I'd guess so - at least if there's still more than one by the time we've processed assetlinks.json, checked the intent filters and checked the signatures.

We're hopefully getting into "that really shouldn't happen" or "if it does happen it really doesn't matter which app is used" territory. I'm not sure what app links do in that scenario.